### PR TITLE
Debug attach

### PR DESCRIPTION
**Pull request by stevemessick**
WIP for using breakpoints in modules. There's at least one line of code that is commented-out that I'm not sure won't be needed, so I added a TODO after it.

DartVmServiceDebugProcess.java got reformatted, making the changes look more substantial than they are. The main change is the addition of lines 555-558.

@pq @devoncarew @jacob314

### CODE REVIEW

**Comment by pq** [comment]
Sorry for the slowness...  Not sure I'm best qualified but happy to take a look if this is still something you want to land?

**Comment by stevemessick** [comment]
Sure, I already have some changes, but it will be easier to review those after this lands.

File: src/io/flutter/server/vmService/DartVmServiceDebugProcess.java [hunk 1]
```diff
@@ -35,78 +39,84 @@
...
   private final ExecutionEnvironment executionEnvironment;
-
   @NotNull
   private final PositionMapper mapper;
-
+  @Nullable protected String myRemoteProjectRootUri;
+  private boolean myVmConnected = false;
   @NotNull private final OpenDartObservatoryUrlAction myOpenObservatoryAction =
     new OpenDartObservatoryUrlAction(null, () -> myVmConnected && !getSession().isStopped());
+  private VmServiceWrapper myVmServiceWrapper;
```
Comment (jacob314) [inline src/io/flutter/server/vmService/DartVmServiceDebugProcess.java]: can you layer the formatting only changes as a separate patch so it is easier to tell what is new what what has moved? I'm having trouble quickly telling what is the diff to review.

File: src/io/flutter/server/vmService/VmServiceWrapper.java [hunk 2]
```diff
@@ -204,6 +208,25 @@ public void received(Success response) {
     }
   }
 
+  public void attachIsolate(@NotNull IsolateRef isolateRef) {
+    //myDebugProcess.getSession().initBreakpoints(); // TODO(messick) Remove if not needed.
```
Comment (jacob314) [inline src/io/flutter/server/vmService/VmServiceWrapper.java]: you are calling initBreakpoints right before calling this method so I expect this isn't needed unless there is a separate code path.

File: src/io/flutter/server/vmService/DartVmServiceDebugProcess.java (hunk 1 above)
Comment (stevemessick) [reply src/io/flutter/server/vmService/DartVmServiceDebugProcess.java]: I initially wrote "The main change is the addition of lines 555-558." I double-checked; that is the only change I intended to make. If you want, I can make a PR that just does the formatting. Once that's merged the diff viewer here will only show lines 555-558.

File: src/io/flutter/server/vmService/VmServiceWrapper.java (hunk 2 above)
Comment (stevemessick) [reply src/io/flutter/server/vmService/VmServiceWrapper.java]: The are two paths to this code, and I think that's true for both.

**Comment by stevemessick** [comment]
Thanks for the review!

**Comment by jacob314** [comment]
Lgtm. I've reviewed that range of the file carefully. You are good to
submit.

**Pull request merged**